c++/include/orc/Type.hh

```diff
@@ -65,6 +67,7 @@ namespace orc {
     virtual std::vector<std::string> getAttributeKeys() const = 0;
     virtual std::string getAttributeValue(const std::string& key) const = 0;
     virtual std::string toString() const = 0;
+    virtual std::optional<const Type*> getTypeByColumnId(uint64_t colId) const = 0;
```

Comment:
I didn’t notice that this is a public header. As Apache impala is still on c++14 and ‘optional’ is a c++17 feature, it will break impala. Could you revert this and add a comment for the possible nullptr of return? Sorry about that.

Reply:
done